src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1275,6 +1278,34 @@ internal static object CopyType(object obj)
         }
 
         #endregion Type
+    
+        #region CancellationTokenWrapper
+        internal static void SerializeCancellationTokenWrapper(object obj, BinaryTokenStreamWriter stream, Type expected)
+        {
+            var ctw = (CancellationTokenWrapper)obj;
+            var cancelled = ctw.CancellationToken.IsCancellationRequested;
+            if (!cancelled)
+            {
+                CancellationTokenManager.RegisterTokenCallbacks(ctw);
```

Comment:
Agreed. Would it be fine to move this methods into `CancellationTokenWrapper` and mark them with  `[SerializerMethod]`, `[DeserializerMethod]` .. attributes?